presto-tests/src/test/java/io/prestosql/execution/TestExecutionJmxMetrics.java

```diff
@@ -0,0 +1,107 @@
+import io.prestosql.spi.QueryId;
+import io.prestosql.testing.DistributedQueryRunner;
+import io.prestosql.tests.tpch.TpchQueryRunnerBuilder;
+import org.testng.annotations.Test;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import static io.prestosql.execution.QueryState.FAILED;
+import static io.prestosql.execution.QueryState.QUEUED;
+import static io.prestosql.execution.QueryState.RUNNING;
+import static io.prestosql.execution.TestQueryRunnerUtil.cancelQuery;
+import static io.prestosql.execution.TestQueryRunnerUtil.createQuery;
+import static io.prestosql.execution.TestQueryRunnerUtil.waitForQueryState;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.assertEquals;
+
+public class TestExecutionJmxMetrics
+{
+    private static final String LONG_RUNNING_QUERY = "SELECT COUNT(*) FROM tpch.sf100000.lineitem";
+
+    @Test(timeOut = 30_000)
+    public void testQueryStats()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = TpchQueryRunnerBuilder.builder().build()) {
+            queryRunner.installPlugin(new ResourceGroupManagerPlugin());
+            InternalResourceGroupManager<?> resourceGroupManager = queryRunner.getCoordinator().getResourceGroupManager()
+                    .orElseThrow(() -> new IllegalStateException("Resource manager not configured"));
+            resourceGroupManager.setConfigurationManager(
+                    "file",
+                    ImmutableMap.of(
+                            "resource-groups.config-file",
+                            getClass().getClassLoader().getResource("resource_groups_single_query.json").getPath()));
+            MBeanServer mbeanServer = queryRunner.getCoordinator().getMbeanServer();
+
+            QueryId firstDashboardQuery = createQuery(queryRunner, dashboardSession(), LONG_RUNNING_QUERY);
+            waitForQueryState(queryRunner, firstDashboardQuery, RUNNING);
+            // the second "dashboard" query can't run right away because the resource group has a hardConcurrencyLimit of 1
+            QueryId secondDashboardQuery = createQuery(queryRunner, dashboardSession(), LONG_RUNNING_QUERY);
+            waitForQueryState(queryRunner, secondDashboardQuery, QUEUED);
+
+            assertEquals(getDispatchManagerAttribute(mbeanServer, "RunningQueryCount"), 1);
+            assertEquals(getDispatchManagerAttribute(mbeanServer, "RunningQueries"), 1);
+            assertEquals(getQueryManagerAttribute(mbeanServer, "RunningQueries"), 1);
+
+            assertEquals(getDispatchManagerAttribute(mbeanServer, "QueuedQueryCount"), 1);
+            // the old attribute has the correct value
+            assertEquals(getDispatchManagerAttribute(mbeanServer, "QueuedQueries"), 1);
+            // this MBean does not track queued queries correctly
+            assertEquals(getQueryManagerAttribute(mbeanServer, "QueuedQueries"), 0);
```

Comment:
I've added these assertions just to illustrate the problem.  I'll remove them once we settle on a way to address it.